app/models/teaching_period.rb

```diff
@@ -0,0 +1,17 @@
+class TeachingPeriod < ActiveRecord::Base
+  has_many :units
+
+  validates :period, length: { minimum: 1, maximum: 20, allow_blank: false }, uniqueness: true
+  validates :year, length: { is: 4, allow_blank: false }, presence: true, numericality: { only_integer: true },
+    inclusion: { in: 2000..2099, message: "%{value} is not a valid year" }
```

Comment:
Lets give us a 1000 year window :) make it 2999

Reply:
Done :+1: